packages/frontend/src/features/semanticViewer/runners/SemanticViewerResultsRunner.ts

```diff
@@ -64,15 +51,74 @@ function getVizIndexTypeFromDimensionType(
         default:
             return assertUnreachable(type, `Unknown field type: ${type}`);
     }
-}
+};
+
+const convertColumnNamesToVizColumns = (
+    fields: SemanticLayerField[],
+    columnNames: string[],
+): VizColumn[] => {
+    return columnNames
+        .map<VizColumn | undefined>((columnName) => {
+            const field = fields.find((f) => f.name === columnName);
+            if (!field) {
+                return;
+            }
+
+            const dimType = getDimensionTypeFromSemanticLayerFieldType(
+                field.type,
+            );
+
+            return {
+                reference: columnName,
+                type: dimType,
+            };
+        })
+        .filter((c): c is VizColumn => Boolean(c));
+};
+
+// This fields dependency should be fixed by fixing the API for semantic layer
+export const getPivotQueryFunctionForSemanticViewer = (
```

Comment:
This function has to vary per runner because we need to refactor more of query execution